tests/installation/select_patterns_and_packages.pm

```diff
@@ -175,25 +176,34 @@ sub run {
         return 1;
     }
     if (get_var('PATTERNS')) {
-        my %wanted_patterns;
+        my %patterns;
         for my $p (split(/,/, get_var('PATTERNS'))) {
-            $wanted_patterns{$p} = 1;
+            $patterns{$p} = 1;
         }
 
         my $counter = 80;
         while (1) {
             die "looping for too long" unless ($counter--);
             my $needs_to_be_selected;
             my $ret = check_screen('on-pattern', 1);
+            my $current_pattern = 'UNKNOWN_PATTERN';
```

Comment:
not sure that I understand purpose of this variable. Looks like you use it to detect if PATTERNS variable has at least one pattern to select and it is not default, right ?